src/wordpress/tests/test_plugins.py

```diff
@@ -0,0 +1,162 @@
+import pytest
+import yaml
+import os
+from settings import PLUGINS_CONFIG_GENERIC_FOLDER, PLUGINS_CONFIG_SPECIFIC_FOLDER
+from wordpress.plugins import WPPluginList, WPPluginConfig, WPConfig
+from wordpress.generator import MockedWPGenerator
+
+TEST_SITE = 'unittest'
+SITE_URL_GENERIC = "https://localhost/"
+SITE_URL_SPECIFIC = "https://localhost/{}".format(TEST_SITE)
+TEST_ENV = 'test'
+
+
+@pytest.fixture(scope="module")
+def wp_plugin_list():
+    return WPPluginList(PLUGINS_CONFIG_GENERIC_FOLDER, 'config-lot1.yml', PLUGINS_CONFIG_SPECIFIC_FOLDER)
+
+
+@pytest.fixture(scope="class")
+def wp_site_generic():
+    # To generate website with generic plugin list/configuration
+    generator = MockedWPGenerator(
+                openshift_env=TEST_ENV,
+                wp_site_url=SITE_URL_GENERIC,
+                wp_default_site_title="My test")
+    generator.clean()
+    generator.generate()
+    return generator.wp_site
+
+
+@pytest.fixture(scope="class")
+def wp_site_specific():
+    # To generate website with specific plugin list/configuration
+    generator = MockedWPGenerator(
+                openshift_env=TEST_ENV,
+                wp_site_url=SITE_URL_SPECIFIC,
+                wp_default_site_title="My test")
+    generator.clean()
+    generator.generate()
+    return generator.wp_site
+
+
+def test_yaml_include():
+    # Generate filename to open regarding current script path
+    yaml_file = os.path.join(os.path.dirname(os.path.realpath(__file__)), 'yaml-root.yml')
+    yaml_content = yaml.load(open(yaml_file, 'r'))
+    assert yaml_content['root_value'] == 'root'
+    assert yaml_content['included_value'] == 'included'
+
+
+class TestWPPluginList:
+
+    def test_generic_plugin_list(self, wp_plugin_list):
+        plugins_to_test = ['add-to-any', 'hello', 'akismet']
+
+        plugin_list = wp_plugin_list.plugins()
+        assert len(plugin_list) == len(plugins_to_test)
+        for plugin_name in plugins_to_test:
+            assert plugin_name in plugin_list
+
+    def test_specific_plugin_list(self, wp_plugin_list):
+        plugins_to_test = ['add-to-any', 'hello', 'epfl_infoscience', 'akismet']
+
+        plugin_list = wp_plugin_list.plugins(TEST_SITE)
+        assert len(plugin_list) == len(plugins_to_test)
+        for plugin_name in plugins_to_test:
+            assert plugin_name in plugin_list
+
+
+class TestWPPluginConfig:
+
+    def test_valid_install_generic(self, wp_site_generic, wp_plugin_list):
+        # Plugins and if they have to be installed or not
+        plugins_to_test = {'add-to-any': True, 'hello': False, 'akismet': True}
+
+        for plugin_name in plugins_to_test:
+
+            plugin_config = wp_plugin_list.plugins()[plugin_name]
+            wp_plugin_config = WPPluginConfig(wp_site_generic, plugin_name, plugin_config)
+
+            assert wp_plugin_config.is_installed is plugins_to_test[plugin_name]
```

Comment:
on test bien un boolean (True ou False) ? Je ne suis pas sûr que `is` soit le bon opérateur. C'est plutôt utilisé pour comparer que [deux objets sont en fait le même](https://docs.python.org/2/reference/expressions.html#is) (et pour `None` aussi). Mais pas pour comparer des valeurs.

Contre exemple:

```
>>> x = [1, 2, 3]
>>> y = [1, 2, 3]
>>> x is y
False
```

avec l'utilisation de items (c.f commentaire ci dessus), la ligne devient:

`assert wp_plugin_config.is_installed == plugin_state`